pkg/segment/query/querystatus.go

```diff
@@ -106,6 +106,7 @@ type RunningQueryState struct {
 	currentSearchResultCount int
 	nodeResult               *structs.NodeResult
 	totalRecsToBeSearched    uint64
+	AllColsInAggs            map[string]struct{}
```

Comment:
I am not exactly sure, where else we could set this value. This probably will not work for synchronous queries. As the query seems to be deleted by the time it reaches the `read raw records` block. Please suggest me if there is a better place where I could store and retrieve this value.

Reply:
you could probably store it in segquery or segsearch

Reply:
Or you could do a call to extract out these columns and just pass it around right from segquery in all the search/apply functions, so that you don't have to depend on this other module which sets/gets it via a lock

Reply:
yeah okay I guess I can do that. Let me check.